### PR TITLE
Add VLAN hack, network-config, UPF first

### PR DESCRIPTION
**Pull request by johnbelamaric**

### CODE REVIEW

**Comment by johnbelamaric** [comment]
/assign @henderiw @electrocucaracha 

005.sh and 006.sh were minimally modified, but their names were switched so it looks like lots of change

File: e2e/provision/hacks/vlan-interfaces.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    echo docker exec $worker ip link add link eth1 name eth1.3 type vlan id 3
+    docker exec $worker ip link add link eth1 name eth1.3 type vlan id 3
+    echo docker exec $worker ip link add link eth1 name eth1.4 type vlan id 4
+    docker exec $worker ip link add link eth1 name eth1.4 type vlan id 4
+    echo docker exec $worker ip link set up eth1.2
+    docker exec $worker ip link set up eth1.2
+    echo docker exec $worker ip link set up eth1.3
+    docker exec $worker ip link set up eth1.3
+    echo docker exec $worker ip link set up eth1.4
+    docker exec $worker ip link set up eth1.4
```
Comment (electrocucaracha) [inline e2e/provision/hacks/vlan-interfaces.sh:36]:
```suggestion
    workers=$(kubectl get nodes -l node-role.kubernetes.io/control-plane!= -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' --kubeconfig "$_kubeconfig")
    for worker in $workers; do;
        for i in {2..4}; do
            docker exec "${worker//\"/}" ip link add link eth1 name "eth1.$i" type vlan id "$i"
            docker exec "${worker//\"/}" ip link set up "eth1.$i"
        done
    done
```

File: e2e/provision/playbooks/roles/install/tasks/main.yml [hunk 2]
```diff
@@ -15,6 +15,7 @@
     - {pkg: porch-dev, version: porch-dev/v2, namespaces: [porch-fn-system, porch-system]}
     - {pkg: nephio-controllers, version: nephio-controllers/v5, namespaces: [nephio-system]}
     - {pkg: configsync, version: configsync/v1, namespaces: [config-management-monitoring, config-management-system, resource-group-system]}
+    - {pkg: network-config, version: network-config/v1, namespaces: []}
```
Comment (electrocucaracha) [inline e2e/provision/playbooks/roles/install/tasks/main.yml:18]: This change will require the addition of a system test [here](https://github.com/nephio-project/test-infra/blob/main/e2e/provision/playbooks/roles/install/molecule/default/tests/test_default.py)

File: e2e/provision/hacks/vlan-interfaces.sh (hunk 1 above)
Comment (electrocucaracha) [reply e2e/provision/hacks/vlan-interfaces.sh:36]: @johnbelamaric the CI is reporting [an error](http://prow.nephio.io/view/gs/prow-nephio-sig-release/pr-logs/pull/nephio-project_test-infra/125/e2e/1673493710991527936#1:build-log.txt%3A52654), this is related to the way workers are retrieved. 

I think that the previous suggestion fixes the issue.

Comment (henderiw) [reply e2e/provision/hacks/vlan-interfaces.sh]: this is the script I used and this worked fine

##############################################################################
 # Copyright (c) 2023 The Nephio Authors.
 # All rights reserved. This program and the accompanying materials
 # are made available under the terms of the Apache License, Version 2.0
 # which accompanies this distribution, and is available at
 # http://www.apache.org/licenses/LICENSE-2.0
 ##############################################################################
 ##
 set -o pipefail
 set -o errexit
 set -o nounset
 [[ ${DEBUG:-false} != "true" ]] || set -o xtrace
 export HOME=${HOME:-/home/ubuntu/}
 export E2EDIR=${E2EDIR:-$HOME/test-infra/e2e}
 export TESTDIR=${TESTDIR:-$E2EDIR/tests}
 export LIBDIR=${LIBDIR:-$E2EDIR/lib}
 source "${LIBDIR}/k8s.sh"
 kubeconfig="$HOME/.kube/config"
 for cluster in $(kubectl get cluster -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' --sort-by=.metadata.name --kubeconfig "$kubeconfig"); do
     _kubeconfig=$(k8s_get_capi_kubeconfig "$kubeconfig" "default" "$cluster")

     workers=$(kubectl get nodes -l node-role.kubernetes.io/control-plane!= -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' --kubeconfig "$_kubeconfig")
     for worker in $workers; do
         for i in {2..4}; do
             docker exec "$worker" ip link add link eth1 name "eth1.$i" type vlan id "$i"
             docker exec "$worker" ip link set up "eth1.$i"
         done
     done
done

**Comment by henderiw** [comment]
we should tag the latest packages and update the files.

003-network.yaml -> should go to v2 once tagged
005-edge-free5gc-upf.yaml -> should go to v4 once tagged
006-regional-free5gc-amf.yaml -> should go to v4 once tagged
006-regional-free5gc-smf.yaml -> should go to v4 once tagged

File: e2e/tests/005.sh [hunk 3]
```diff
@@ -27,20 +27,18 @@ source "${LIBDIR}/k8s.sh"
...
+done
 
-# check the SMF
-k8s_wait_ready "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-smf-regional-free5gc-smf"
-k8s_wait_exists "$kubeconfig" 600 "default" "packagevariant" "regional-free5gc-smf-regional-free5gc-smf"
-k8s_wait_exists "$cluster_kubeconfig" 600 "free5gc-cp" "deployment" "smf-regional"
-k8s_wait_ready_replicas "$cluster_kubeconfig" 600 "free5gc-cp" "deployment" "smf-regional"
+for cluster in "edge01" "edge02"; do
+    cluster_kubeconfig=$(k8s_get_capi_kubeconfig "$kubeconfig" "default" "$cluster")
+    k8s_wait_exists "$cluster_kubeconfig" 600 "free5gc-upf" "deployment" "free5gc-upf"
```
Comment (henderiw) [inline e2e/tests/005.sh:42]: k8s_wait_exists "$cluster_kubeconfig" 600 "free5gc-upf" "deployment" "upf-${cluster}"

**Review by henderiw — CHANGES_REQUESTED** [review]

File: e2e/tests/005.sh (hunk 3 above)
Comment (henderiw) [reply e2e/tests/005.sh:42]: this fixed the issue for me with the checks

Comment (johnbelamaric) [reply e2e/tests/005.sh:42]: yes, I pushed that fix, thanks. So close :)

**Comment by henderiw** [comment]
one thing I noticed is this pr does not contain the latest changes on 003 additions we did

File: e2e/provision/playbooks/roles/install/tasks/main.yml (hunk 2 above)
Comment (johnbelamaric) [reply e2e/provision/playbooks/roles/install/tasks/main.yml:18]: Done, thanks!

**Comment by henderiw** [comment]
I have to chmod +x ../provision/hacks/vlan-interfaces.sh to run the 002.sh

**Comment by johnbelamaric** [comment]
> I have to chmod +x ../provision/hacks/vlan-interfaces.sh to run the 002.sh

Thanks, fixed

**Pull request merged**